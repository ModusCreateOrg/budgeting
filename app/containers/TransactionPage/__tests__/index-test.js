import React from 'react';
import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router';
import { TransactionPage } from '../index';

// mock nested component
jest.mock('containers/TransactionDetails');

it('renders correctly', () => {
  const mockTransaction = {
    id: 1,
    description: "Trader Joe's food",
    value: -423.34,
    categoryId: '1',
  };

  const tree = renderer
    .create(
      <MemoryRouter>
        <TransactionPage transaction={mockTransaction} />
      </MemoryRouter>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it(`renders with message 'transaction not found' if transaction is 'undefined'`, () => {
  const tree = renderer
    .create(
      <MemoryRouter>
        <TransactionPage />
      </MemoryRouter>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
