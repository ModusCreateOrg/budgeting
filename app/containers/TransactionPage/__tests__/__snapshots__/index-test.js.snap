// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <a
    href="/budget"
    onClick={[Function]}
  >
    ← Back to list
  </a>
</div>
`;

exports[`renders with message 'transaction not found' if transaction is 'undefined' 1`] = `
<div>
  <a
    href="/budget"
    onClick={[Function]}
  >
    ← Back to list
  </a>
  <p
    className="notFoundMsg"
  >
    Transaction could not be found!
  </p>
</div>
`;
